geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/wancommand/ClusterConfigStartStopPauseAndResumeGatewaySenderOperationDUnitTest.java

```diff
@@ -0,0 +1,716 @@
+  }
+
+  /**
+   * Verify that parallel gateway-sender queue recovers only data enqueued prior to stop command
+   * from disk-store.
+   *
+   * - Region type: PARTITION_PERSISTENT and non-redundant
+   * - Gateway sender configured with queue persistence
+   *
+   * 1. Pause gateway-sender
+   * 2. Run some traffic and verify that data is enqueued in gateway-sender queues
+   * 3. Stop gateway-sender
+   * 4. Run some traffic and verify that data is stored in region, and not enqueued
+   * 6. Restart all servers
+   * 7. Check that data that is enqueued prior to stop command is recovered from persistent storage,
+   * and that gateway-sender remained in stopped state
+   * 8. Start gateway-senders
+   * 9. Check that data is not replicated to remote site
+   */
+  @Test
+  @Parameters({"false", "true"})
+  public void testThatStopStateRemainAfterTheRestartAndQueueDataIsRecovered(String isParallel)
+      throws Exception {
+    configureSites(true, "PARTITION_PERSISTENT", "0", isParallel);
+
+    executeGfshCommand(CliStrings.PAUSE_GATEWAYSENDER);
+    verifyGatewaySenderState(true, true);
+    server2Site2.invoke(() -> waitAllDispatcherThreadsToPause("ln"));
+    server1Site2.invoke(() -> waitAllDispatcherThreadsToPause("ln"));
+
+    Set<String> keysQueued = clientSite2.invoke(() -> doPutsInRange(70, 85));
+    clientSite2.invoke(() -> checkDataAvailable(keysQueued));
+    server1Site2.invoke(() -> checkQueueSize("ln", keysQueued.size()));
+
+    executeGfshCommand(CliStrings.STOP_GATEWAYSENDER);
+
+    verifyGatewaySenderState(false, isParallel.equals("true"));
+
+    Set<String> keysNotQueued = clientSite2.invoke(() -> doPutsInRange(100, 105));
+    clientSite2.invoke(() -> checkDataAvailable(keysNotQueued));
+
+    // stop servers on site #2
+    server1Site2.stop(false);
+    server2Site2.stop(false);
+
+    // start again servers in Site #2
+    Thread thread = new Thread(
+        () -> server1Site2 = clusterStartupRule.startServerVM(5, locatorSite2.getPort()));
+    Thread thread1 = new Thread(
+        () -> server2Site2 = clusterStartupRule.startServerVM(6, locatorSite2.getPort()));
+    // start threads
+    thread.start();
+    thread1.start();
+    thread.join();
+    thread1.join();
+
+    verifyGatewaySenderState(false, false);
+    if (isParallel.equals("true")) {
+      server1Site2.invoke(() -> checkQueueSize("ln", keysQueued.size()));
+    }
+    executeGfshCommand(CliStrings.START_GATEWAYSENDER);
+    verifyGatewaySenderState(true, false);
+
+    // Check that data is sent over the gateway-sender
+    server1Site2.invoke(() -> checkQueueSize("ln", 0));
+    clientSite1.invoke(() -> checkDataAvailable(keysQueued));
+    clientSite1.invoke(() -> checkDataNotAvailable(keysNotQueued));
+  }
+
+  /**
+   * This test case verifies that the server during recovery of gateway-sender in stopped state
+   * stores events in tmpDroppedEvents queue, and handles them after sender has been recovered.
+   */
```

Comment:
Would it be possible to check that tmpDroppedEvents is not zero at some point of the test and then, at the end, it is zero?
Otherwise, I do not think that the test case does what it should.